pkg/services/ngalert/state/cache.go

```diff
@@ -35,12 +36,13 @@ func (c *cache) getOrCreate(alertRule *ngModels.AlertRule, result eval.Result) *
 	c.mtxStates.Lock()
 	defer c.mtxStates.Unlock()
 
-	templateData := make(map[string]string, len(result.Instance)+3)
+	labels := make(map[string]string)
```

Comment:
```suggestion
	labels := result.Instance.Copy()
```
nit: `Labels` have a method to copy themselves.